libraries/AP_Networking/AP_Networking.cpp

```diff
@@ -0,0 +1,393 @@
+    // @Range: 0 255
+    // @RebootRequired: True
+    // @User: Advanced
+    AP_GROUPINFO("GWADDR1", 8,  AP_Networking,    _param.gwaddr[1],   AP_NETWORKING_DEFAULT_STATIC_GW_ADDR1),
+
+    // @Param: GWADDR2
+    // @DisplayName: Gateway IP Address 3rd byte
+    // @Description: Allows setting static GW address
+    // @Range: 0 255
+    // @RebootRequired: True
+    // @User: Advanced
+    AP_GROUPINFO("GWADDR2", 9,  AP_Networking,    _param.gwaddr[2],   AP_NETWORKING_DEFAULT_STATIC_GW_ADDR2),
+
+    // @Param: GWADDR3
+    // @DisplayName: Gateway IP Address LSB
+    // @Description: Allows setting static GW address
+    // @Range: 0 255
+    // @RebootRequired: True
+    // @User: Advanced
+    AP_GROUPINFO("GWADDR3", 10,  AP_Networking,    _param.gwaddr[3],   AP_NETWORKING_DEFAULT_STATIC_GW_ADDR3),
+
+    // @Param: MACADDR0
+    // @DisplayName: MAC Address MSbyte
+    // @Description: Allows setting MAC address
+    // @Range: 0 255
+    // @RebootRequired: True
+    // @User: Advanced
+    AP_GROUPINFO("MACADDR0", 11,  AP_Networking,    _param.macaddr[0],  AP_NETWORKING_DEFAULT_MAC_ADDR0),
+
+    // @Param: MACADDR1
+    // @DisplayName: MAC Address 2nd byte
+    // @Description: Allows setting MAC address
+    // @Range: 0 255
+    // @RebootRequired: True
+    // @User: Advanced
+    AP_GROUPINFO("MACADDR1", 12,  AP_Networking,    _param.macaddr[1],   AP_NETWORKING_DEFAULT_MAC_ADDR1),
+
+    // @Param: MACADDR2
+    // @DisplayName: MAC Address 3rd byte
+    // @Description: Allows setting MAC address
+    // @Range: 0 255
+    // @RebootRequired: True
+    // @User: Advanced
+    AP_GROUPINFO("MACADDR2", 13,  AP_Networking,    _param.macaddr[2],   AP_NETWORKING_DEFAULT_MAC_ADDR2),
+
+    // @Param: MACADDR3
+    // @DisplayName: MAC Address 4th byte
+    // @Description: Allows setting MAC address
+    // @Range: 0 255
+    // @RebootRequired: True
+    // @User: Advanced
+    AP_GROUPINFO("MACADDR3", 14,  AP_Networking,    _param.macaddr[3],   AP_NETWORKING_DEFAULT_MAC_ADDR3),
+
+    // @Param: MACADDR4
+    // @DisplayName: MAC Address 5th byte
+    // @Description: Allows setting MAC address
+    // @Range: 0 255
+    // @RebootRequired: True
+    // @User: Advanced
+    AP_GROUPINFO("MACADDR4", 15,  AP_Networking,    _param.macaddr[4],   AP_NETWORKING_DEFAULT_MAC_ADDR4),
+
+    // @Param: MACADDR5
+    // @DisplayName: MAC Address LSb
+    // @Description: Allows setting MAC address
+    // @Range: 0 255
+    // @RebootRequired: True
+    // @User: Advanced
+    AP_GROUPINFO("MACADDR5", 16,  AP_Networking,    _param.macaddr[5],   AP_NETWORKING_DEFAULT_MAC_ADDR5),
+
+    AP_GROUPEND
+};
+
+AP_Networking::AP_Networking(void)
+{
+#if CONFIG_HAL_BOARD == HAL_BOARD_SITL
+    if (_singleton != nullptr) {
+        AP_HAL::panic("AP_Networking must be singleton");
+    }
+#endif
+
+    _singleton = this;
+    AP_Param::setup_object_defaults(this, var_info);
+}
+
+
+void AP_Networking::init()
```

Comment:
Any reason you made this void return and swallow the error instead of  return a `bool` for success/fail  to propogate the error to the caller?

Reply:
All the inits are like this. The caller doesn't have any error handling. The module itself must handle then

Reply:
For example, here's where it and a bunch of other libraries are calling init at boot
https://github.com/ArduPilot/ardupilot/blob/master/libraries/AP_Vehicle/AP_Vehicle.cpp#L271

Keep in mind, errors are not treated the same in the embedded world vs desktop OS world